### PR TITLE
WIP:allow commenting in different languages. Implements #422

### PR DESCRIPTION
**Pull request by adsworth**
when switching language in the rating form loads the current users comment in the newly selected language. What is open is showing a dialog, if there is unsaved data, when switching language asking the user if he want's to save.

### CODE REVIEW

File: nextcloudappstore/core/static/assets/js/ui/app-detail.js [hunk 1]
```diff
@@ -41,6 +41,16 @@
...
     }
 
+    function load_language(lang) {
+        commentTextarea.readOnly = true;
+        fetch(commentUrl + "?lang=" + lang, {credentials: 'include'})
+            .then((response) => response.json())
+            .then((json) => {
+                commentTextarea.value = json.comment;
+        });
+        commentTextarea.readOnly = false;
```
Comment (BernhardPosselt) [inline nextcloudappstore/core/static/assets/js/ui/app-detail.js]: This part will execute after you fired the request, not when it finished successfully, see https://github.com/github/fetch/issues/223#issuecomment-148927226

File: nextcloudappstore/core/views.py [hunk 2]
```diff
@@ -45,6 +52,24 @@ def get_queryset(self):
             app=app)
 
 
+class AppCommentApi(View):
+
+    def get(self, request, *args, **kwargs):
```
Comment (BernhardPosselt) [inline nextcloudappstore/core/views.py]: I'm unsure if you can just extend a DetailView and reuse more code, just take a look ;)

**Comment by BernhardPosselt** [comment]
One thing I was wondering about btw: if a user changes the language drop down and he changed the content, should we ask him to confirm to drop changes?

**Comment by adsworth** [comment]
I guess we have to ask for some kind of cinformation, otherwise we are going to loose data.
Can you point me somewhere so I can implement a nice modal popup asking for confirmation.
And then the question is do we submit the new comment via an API or do we just let the user submit the form?

**Comment by BernhardPosselt** [comment]
As for modals: simply use the built in (confirm()), shouldn't be to bad in this case :)
Comment submission should be via plain form

**Comment by BernhardPosselt** [comment]
Moved most of your code into https://github.com/nextcloud/appstore/pull/455/files 

Let's not brood too much about the current situation, let's first get in a working solution and work on UX later once we've got a good enough idea of how we should actually implement it.

**Pull request closed without merging**